### PR TITLE
APPS/IODEMO: Backport PR#7834

### PR DESCRIPTION
**Pull request by huaxiangfan**
Backport https://github.com/openucx/ucx/pull/7834

### CODE REVIEW

**Comment by yosefe** [comment]
@huaxiangfan can you pls look at the merge conflict?

**Comment by huaxiangfan** [comment]
> @huaxiangfan can you pls look at the merge conflict?

done

File: test/apps/iodemo/io_demo.cc [hunk 1]
```diff
@@ -2392,6 +2408,8 @@ static int parse_args(int argc, char **argv, options_t *test_opts)
             break;
         case 'z':
             test_opts->prereg = true;
+        case 'V':
```
Comment (yosefe) [inline test/apps/iodemo/io_demo.cc:2412]: missing "break"

Comment (huaxiangfan) [reply test/apps/iodemo/io_demo.cc:2412]: done

**Review by yosefe — APPROVED** [review]
pls squash and rebase

**Comment by huaxiangfan** [comment]
> pls squash and rebase

done

**Review by yosefe — APPROVED** [review]

**Pull request merged**